### PR TITLE
Convert storage units component of a PyPSA Network object to storage component of a Grid object

### PR DESCRIPTION
**Pull request by chrstphtrs**
[Pull Request doc](https://breakthrough-energy.github.io/docs/user/git_guide.html#d-pull-request)

### Purpose
What is the larger goal of this change?
Add conversion of PyPSA "network.storage_units" to PowerSimData "grid.storages" dictionary in [powersimdata/input/converter/pypsa_to_grid.py](https://github.com/Breakthrough-Energy/PowerSimData/commit/704722c5e51e32ba040551cd67b3e83c459f30dd#diff-5ad2d8a809c6a0852ef673b4f590447bdc37e7e295d4d841877429cbd97c4570)

### What the code is doing
How is the purpose executed?
Handle columns from PyPSA's "network.storage_units" to match columns for each of the three dataframes within the "grid.storages" dictionary ("gen", "gencost", "StorageData"). This operation is performed analogously to plant/bus etc. dataframes: (1) Rename columns with matching data, (2) create new columns for more complex relationships, (3) drop redundant columns with information from PyPSA that is not needed in PowerSimData. Finally, the "grid.storages" dictionary is put together from the three dataframes.

### Testing
How did you test this change (unit/functional testing, manual testing, etc.)? 
Built additional test for a PyPSA network including storages to be converted into a PowerSimData grid (in [powersimdata/input/converter/tests/test_pypsa_to_grid.py](https://github.com/Breakthrough-Energy/PowerSimData/commit/704722c5e51e32ba040551cd67b3e83c459f30dd#diff-2d590f57b8c502c42531512361a746d773531e9ea2a35e5366546f2481fda1b3)). Manually ran this test individually, as well as the entire test_pypsa_to_grid.py. Then, ran full tox tests.

### Where to look
* It's helpful to clarify where your new code lives if you moved files around or there could be confusion/
- [powersimdata/input/converter/pypsa_to_grid.py](https://github.com/Breakthrough-Energy/PowerSimData/commit/704722c5e51e32ba040551cd67b3e83c459f30dd#diff-5ad2d8a809c6a0852ef673b4f590447bdc37e7e295d4d841877429cbd97c4570)
- [powersimdata/input/exporter/export_to_pypsa.py](https://github.com/Breakthrough-Energy/PowerSimData/commit/704722c5e51e32ba040551cd67b3e83c459f30dd#diff-41147490cd9f763f07291cb89af5d67ddd952bf6ee3a67f25f8413003d63876f)
- [powersimdata/input/converter/tests/test_pypsa_to_grid.py](https://github.com/Breakthrough-Energy/PowerSimData/commit/704722c5e51e32ba040551cd67b3e83c459f30dd#diff-2d590f57b8c502c42531512361a746d773531e9ea2a35e5366546f2481fda1b3)

* What files are most important?
- [powersimdata/input/converter/pypsa_to_grid.py](https://github.com/Breakthrough-Energy/PowerSimData/commit/704722c5e51e32ba040551cd67b3e83c459f30dd#diff-5ad2d8a809c6a0852ef673b4f590447bdc37e7e295d4d841877429cbd97c4570)

### Usage Example/Visuals
How the code can be used and/or images of any graphs, tables or other visuals (not always applicable). 

### Time estimate
How long will it take for reviewers and observers to understand this code change?
15 min

### CODE REVIEW

File: powersimdata/input/converter/pypsa_to_grid.py [hunk 1]
```diff
@@ -3,10 +3,11 @@
 import numpy as np
 import pandas as pd
 
-from powersimdata.input.abstract_grid import AbstractGrid
+from powersimdata.input.abstract_grid import AbstractGrid, storage_template
```
Comment (rouille) [inline powersimdata/input/converter/pypsa_to_grid.py:6]: I believe the storage template is accessible from the `AbstractGrid` object:
```
>>> from powersimdata.input.abstract_grid import AbstractGrid
>>> ag = AbstractGrid()
>>> ag.storage
{'gen': Empty DataFrame
Columns: [bus_id, Pg, Qg, Qmax, Qmin, Vg, mBase, status, Pmax, Pmin, Pc1, Pc2, Qc1min, Qc1max, Qc2min, Qc2max, ramp_agc, ramp_10, ramp_30, ramp_q, apf, mu_Pmax, mu_Pmin, mu_Qmax, mu_Qmin]
Index: [], 'gencost': Empty DataFrame
Columns: [type, startup, shutdown, n, c2, c1, c0]
Index: [], 'StorageData': Empty DataFrame
Columns: [UnitIdx, InitialStorage, InitialStorageLowerBound, InitialStorageUpperBound, InitialStorageCost, TerminalStoragePrice, MinStorageLevel, MaxStorageLevel, OutEff, InEff, LossFactor, rho, ExpectedTerminalStorageMax, ExpectedTerminalStorageMin]
Index: [], 'genfuel': [], 'duration': None, 'min_stor': None, 'max_stor': None, 'InEff': None, 'OutEff': None, 'LossFactor': None, 'energy_price': None, 'terminal_min': None, 'terminal_max': None}
>>> 
```
and there is no need to import `storage_template`

**Comment by jenhagg** [comment]
It seemed like we could break up the creation of storage data frames, so I tried it out in this [branch](https://github.com/Breakthrough-Energy/PowerSimData/compare/ben/import...jen/pr-657?expand=1).

**Comment by rouille** [comment]
What are the purpose of the ***.syncignore-receive*** and ***.syncignore-send*** files? Are these to be used with the `rsync` command with the `--exclude` option? I am not sure we want these files files in the repository but they could be added to the ***.gitignore*** if you want to keep them locally.

**Comment by chrstphtrs** [comment]
Yes, they are for rsync file transfer. Will add to .gitignore and keep them locally, no problem.

**Review by BainanXia — COMMENTED** [review]
Looks good to me in general. Some doc strings are missing which is minor. We use 'PSD' in doc strings to refer `PowerSimData`, just wondering whether we would like to make it an official abbreviation?

Maybe out of the scope of this PR: we are migrating all entries from both `storage_units` and `stores` from a network object into the storage data frame of a grid object here. I'm wondering how to pair newly added hydro plant (pending now) with these storage devices in the upcoming PR?

**Comment by rouille** [comment]
@chrstphtrs, can you perform an interactive rebase to reword your commit message following our [guide](https://breakthrough-energy.github.io/docs/user/git_guide.html#commit-message). Thanks! Otherwise, besides the few docstrings issues pointed out by @BainanXia, it looks good. We look forward to integrate this feature.

**Review by rouille — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**